packages/server/src/api/controllers/query/index.ts

```diff
@@ -311,8 +311,8 @@ export async function preview(
 
   // if existing schema, update to include any previous schema keys
   if (existingSchema) {
-    for (let key of Object.keys(previewSchema)) {
-      if (existingSchema[key]) {
+    for (let key of Object.keys(existingSchema)) {
```

Comment:
This does seem like quite a silly mistake, logically I can see how this was completely wrong - is there anyway we could put a quick test together for this to confirm that it works? This feels like a logical mistake which could easily regress.